### PR TITLE
Add 3D Particles Documentation

### PR DESCRIPTION
**Pull request by and-rad**
This adds a new page to the documentation, covering 3D particle systems.

- [x] Emitter properties
- [x] ParticlesMaterial properties
  - [x] [Update](https://github.com/godotengine/godot/pull/55399): sphere surface
- [x] Sub emitters
- [x] Trails
  - [x] Ribbons
  - [x] Tubes
  - [x] ~Skinned meshes~ ([buggy](https://github.com/godotengine/godot/issues/62170), will be handled in separate PR)
- [x] Collision
  - [x] Box
  - [x] Heightfield
  - [x] SDF
  - [x] Sphere 
- [x] Attractors
  - [x] Box
  - [x] Sphere
  - [x] Vector field
- [x] Complex emission shapes
- [x] Turbulence ([waiting to be merged](https://github.com/godotengine/godot/pull/55387) and [maybe improved](https://github.com/godotengine/godot/pull/64606))
- [x] Viewport menu
  - [x] Create emission points from node
  - [x] Convert to CPUParticles3D
- [x] Split docs into multiple pages

There is also this companion project that I used to create every screenshot used in the documentation:
https://github.com/and-rad/GodotDocParticles

### CODE REVIEW

**Review by clayjohn — COMMENTED** [review]
You are definitely on the right track! This is looking great so far. I'm looking forward seeing to the rest of your progress

**Comment by and-rad** [comment]
Small heads up: It might look like I'm not making any progress lately, but that's just me waiting for 4.0 to become a little more feature-complete. I've had to redo some screenshots and gifs multiple times already since I started writing the docs because names and features have changed in the meantime. And with https://github.com/godotengine/godot-proposals/issues/3588 probably getting implemented, I'll have to reorganize a lot of what's already written.

So in an effort to not have to do too many do-overs, I'm keeping the particle docs on the back burner, but I'll pick up the pace again as soon as 4.0 nears RC status.

**Comment by clayjohn** [comment]
@and-rad Sounds great! Thanks for letting us know. I look forward to when you return to this PR :)

**Comment by and-rad** [comment]
Gonna link https://github.com/godotengine/godot/issues/61014 here. Even though it doesn't currently work, I have documented the feature as if it did. I expect it'll be fixed in time for the 4.0 release.

**Comment by Calinou** [comment]
> Gonna link [godotengine/godot#61014](https://github.com/godotengine/godot/issues/61014) here. Even though it doesn't currently work, I have documented the feature as if it did. I expect it'll be fixed in time for the 4.0 release.

This wasn't fixed for 4.0's release, so I suggest adding a ``warning`` block to mention that the feature won't work yet.

**Review by Calinou — CHANGES_REQUESTED** [review]
The text is very good and complete, thanks a lot for going through this effort!

There are some changes that should be made to page and reference names though. Page names should use `snake_case` as per convention, and reference names should follow the file names (with parents included if there's possible ambiguity with other pages in the documentation).

Some page names should also be changed for better SEO, and so that they make more sense when viewed individually.

Here's a list of proposed changes (old → new):

- `complexshapes.rst` → `complex_shapes.rst`
- `doc_complexemissionshapes3d` → `doc_3d_particles_complex_shapes`
- `creation.rst` → `creating_a_3d_particle_system.rst`
- `doc_particlecreation3d` → `doc_creating_3d_particle_system`
- `doc_particles3d` → `doc_3d_particles`
- `doc_particleprocessmaterial3d` → `doc_process_material_properties`
- `doc_particleattractors3d` → `doc_3d_particle_attractors` (also rename the page to "3D particle attractors", and add a note that 2D particle attractors aren't implemented yet)
- `doc_particlecollision3d` → `doc_particle_collision_3d`
- `doc_particleproperties3d` → `doc_particle_properties_3d`
- `doc_subemitters3d` → `doc_subemitters_3d` (also rename the page to "Particle sub-emitters" with a hyphen)
- `doc_particletrails3d` → `doc_particle_trails_3d`
- `doc_particleturbulence3d` → `doc_particle_turbulence_3d` (also rename the page to "Particle turbulence"

**Comment by and-rad** [comment]
Done. I deviated from some of your suggestions when I felt it would improve consistency, but I think I got the important parts right.

**Review by Calinou — APPROVED** [review]
Thanks a ton!

File: tutorials/3d/particles/turbulence.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+   :align: right
+
+   Particle turbulence properties
+
+There are two things you have to do before turbulence has any effect on a particle system. First you must
+add movement to the particle system. Turbulence modifies a particle's movement
+direction and speed, but it doesn't create any. It is enough to give the particle system some
+gravity, but you can just as well create a number of attractors if you want the particles
+to follow a more complex movement path. Second, you need to :ref:`enable turbulence in the particle process material <doc_process_material_properties_turbulence>`.
+Once enabled, you have access to all the turbulence properties.
```
Comment (Calinou) [inline tutorials/3d/particles/turbulence.rst:24]:
```suggestion
Once enabled, you have access to all the turbulence properties.

.. warning::

	Turbulence has a high performance cost on the GPU. Only enable turbulence on a few particle systems on screen at most. Using turbulence is not recommended when targeting mobile/web platforms.
```

**Comment by Calinou** [comment]
I've amended a PR to document performance caveats of turbulence as per https://github.com/godotengine/godot/issues/76501.

**Review by Calinou — APPROVED** [review]
Thanks a ton!

**Pull request merged**